pyflakes/checker.py

```diff
@@ -870,6 +922,49 @@ def ignore(self, node):
     # additional node types
     COMPREHENSION = KEYWORD = FORMATTEDVALUE = handleChildren
 
+    def DICT(self, node):
+        # Complain if there are duplicate keys with different values
+        # If they have the same value it's not going to cause potentially
+        # unexpected behaviour so we'll not complain.
+        keys = [
+            convert_to_value(key) for key in node.keys
+        ]
+        for key in set(keys):
+            if keys.count(key) <= 1:
+                continue
+            candidate_values = []
+            for candidate in range(len(keys)):
+                if key == keys[candidate]:
+                    candidate_values.append(
+                        convert_to_value(node.values[candidate])
+                    )
+
+            # Determine whether we have different values for the repeats
+            differing_values = False
+            for value in set(candidate_values):
+                if candidate_values.count(value) == 1:
+                    differing_values = True
+                    break
+
+            if differing_values:
```

Comment:
The whole differing_values section can be reduced to a single `if [any|all](): continue` , which reduces the complexity of the method by ~16%